### PR TITLE
SFR-5798: added another type-hinting, return types declarations

### PR DESCRIPTION
**Pull request by pawel-wrzeszcz**

### CODE REVIEW

File: Process/Coordinator/Coordinator.php [hunk 1]
```diff
@@ -103,13 +103,7 @@ public function forward(string $scenarioAlias, string $stepName): Response
...
     }
 
-    /**
-     * @param ProcessInterface $process
-     * @param $result
-     *
-     * @return mixed
-     */
-    public function processStepResult(ProcessInterface $process, $result): mixed
+    public function processStepResult(ProcessInterface $process, mixed $result): Response|RedirectResponse
```
Comment (marek-binkowski-sim) [inline Process/Coordinator/Coordinator.php]: RedirectResponse extends Response, so you don't need to alter them, it is enough it you specify Response

File: Process/Step/AbstractControllerStep.php [hunk 2]
```diff
@@ -23,55 +24,53 @@ abstract class AbstractControllerStep extends Controller implements StepInterfac
...
     /**
      * {@inheritdoc}
      */
-    public function setName($name)
+    public function setName($name): void
     {
         $this->name = $name;
     }
 
     /**
```
Comment (marek-binkowski-sim) [inline Process/Step/AbstractControllerStep.php]: I guess @inheritdoc is not needed anymore either

File: Process/Coordinator/Coordinator.php (hunk 1 above)
Comment (pawel-wrzeszcz) [reply Process/Coordinator/Coordinator.php]: fixed

File: Process/Step/AbstractControllerStep.php (hunk 2 above)
Comment (pawel-wrzeszcz) [reply Process/Step/AbstractControllerStep.php]: fixed

**Pull request merged**